### PR TITLE
fix: fixing the wrong passing of arguments to the parse_file endpoint

### PR DESCRIPTION
**Pull request by jacopo-chevallard**

### CODE REVIEW

File: megaparse/sdk/megaparse_sdk/endpoints/file_upload.py [hunk 1]
```diff
@@ -22,10 +23,10 @@ async def upload(
         with open(file_path, "rb") as file:
             files = {"file": (file_path, file)}
             data = {
-                "method": method,
-                "strategy": strategy,
+                "method": method.value if isinstance(method, Enum) else method,
```
Comment (AmineDiro) [inline megaparse/sdk/megaparse_sdk/endpoints/file_upload.py:26]:
```suggestion
json.dumps(data)
```

File: megaparse/api/app.py [hunk 2]
```diff
@@ -46,20 +47,20 @@ def _check_free_memory() -> bool:
 @app.post("/v1/file")
 async def parse_file(
     file: UploadFile = File(...),
-    method: ParserType = ParserType.UNSTRUCTURED,
-    strategy: StrategyEnum = StrategyEnum.AUTO,
-    check_table=False,
-    language: Language = Language.ENGLISH,
-    parsing_instruction: str | None = None,
-    model_name: str | None = None,
+    method: ParserType = Form(ParserType.UNSTRUCTURED),
```
Comment (AmineDiro) [inline megaparse/api/app.py:50]: We should check if the FastAPI enforces type checks on the form struct using pydantic

File: tests/test_import.py [hunk 3]
```diff
@@ -1,13 +1,29 @@
...
-    response = megaparse.load("./tests/data/files/dummy.pdf")
+    response = megaparse.load("./tests/data/dummy.pdf")
     print(response)
-    assert response.strip("\n") == "# Dummy PDF download"
+    assert response.strip("\n") == "Dummy PDF download"
+
+
+# @pytest.mark.skip("slow test")
+@pytest.mark.asyncio
+async def test_load_SDK():
```
Comment (AmineDiro) [inline tests/test_import.py:21]: Is this a unit test? an integration test ? a performance test?

File: tests/test_endpoints.py
Comment (AmineDiro) [inline tests/test_endpoints.py:1]: We should have extensive tests for all combinaison of (file, config) the API gives us. I think we can set the bar to have at least 90% code coverage for megaparse here

File: megaparse/api/app.py [hunk 4]
```diff
@@ -46,20 +48,20 @@ def _check_free_memory() -> bool:
...
+    strategy: StrategyEnum = Form(StrategyEnum.AUTO),
+    check_table: bool = Form(False),
+    language: Language = Form(Language.ENGLISH),
+    parsing_instruction: Optional[str] = Form(None),
+    model_name: Optional[str] = Form("gpt-4o"),
```
Comment (AmineDiro) [inline megaparse/api/app.py:56]: Do we have a list of model names?

File: megaparse/sdk/megaparse_sdk/endpoints/file_upload.py [hunk 5]
```diff
@@ -1,7 +1,9 @@
 from typing import Optional
 
 from httpx import Response
+
 from megaparse.sdk.megaparse_sdk.client import MegaParseClient
+from megaparse.sdk.megaparse_sdk.config import UploadFileInput
```
Comment (AmineDiro) [inline megaparse/sdk/megaparse_sdk/endpoints/file_upload.py:6]: THis is is more of a config to the parser than file input

File: tests/test_endpoints.py
Comment (chloedia) [reply tests/test_endpoints.py:1]: Yes, I am going to do that in a following PR, I don't want to put everything in the same PR, or should I ?

File: megaparse/sdk/megaparse_sdk/endpoints/file_upload.py (hunk 1 above)
Comment (chloedia) [reply megaparse/sdk/megaparse_sdk/endpoints/file_upload.py]: Done

File: tests/test_import.py (hunk 3 above)
Comment (chloedia) [reply tests/test_import.py:21]: Integration test ? 
What should I modify ?

File: megaparse/api/app.py (hunk 4 above)
Comment (chloedia) [reply megaparse/api/app.py]: Just added it in type.py

File: megaparse/sdk/megaparse_sdk/endpoints/file_upload.py (hunk 5 above)
Comment (chloedia) [reply megaparse/sdk/megaparse_sdk/endpoints/file_upload.py]: Changed the name :)

**Review by AmineDiro — APPROVED** [review]

**Pull request merged**